src/main/java/uk/gov/hmcts/ccd/domain/service/casedataaccesscontrol/RoleToAccessProfilesMappingsGenerator.java

```diff
@@ -0,0 +1,126 @@
+package uk.gov.hmcts.ccd.domain.service.casedataaccesscontrol;
+
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Component;
+import uk.gov.hmcts.ccd.data.caseaccess.CachedCaseRoleRepository;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseRoleRepository;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.AccessProfile;
+import uk.gov.hmcts.ccd.domain.model.definition.AccessControlList;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.ccd.data.caseaccess.GlobalCaseRole.CREATOR;
+
+@Component
+public class RoleToAccessProfilesMappingsGenerator {
+
+    protected static final String IDAM_PREFIX = "idam:";
+    private static final String CASE_ROLE_ID_REGEX = "^(\\[[A-Za-z]+\\])$";
+
+    private final CaseRoleRepository caseRoleRepository;
+
+    public RoleToAccessProfilesMappingsGenerator(@Qualifier(CachedCaseRoleRepository.QUALIFIER)
+                                                     CaseRoleRepository caseRoleRepository) {
+        this.caseRoleRepository = caseRoleRepository;
+    }
+
+    public List<AccessProfile> generate(CaseTypeDefinition caseTypeDefinition) {
+
+        Set<String> caseRoles = generateCaseRoles(caseTypeDefinition);
+        Set<String> idamRoles = generateIdamRoles(caseTypeDefinition);
+
+        List<AccessProfile> accessProfiles = new ArrayList<>();
```

Comment:
this naming is misleading. It seems to me `AccessProfile` is in reality modelling the mapping from role to access profile, not the access profile itself as the name would suggest. 
I think we need to rename this class to `AccessProfileMapping`. Then we need to introduce a separate class, `AccessProfile`, that models the mapped access profiles. With properties like `readOnly`, `securityClassification`...